### PR TITLE
Hwrdtm/datil dev support

### PR DESCRIPTION
**Pull request by hwrdtm**
# Description

Work towards LIT-3404.

This PR:
- Implements changes to support `datil-dev` as a Lit network, both in the core packages and also in the `local-tests` (Tinny). **More importantly, this PR is INCOMPLETE ⚠️ ⚠️ ⚠️**. It currently uses a hack to load in the network context for running against Datil-dev.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Run `DEBUG=true NETWORK=localchain LIT_RPC_URL=https://vesuvius-rpc.litprotocol.com yarn test:local` to run against Datil-dev. 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Datil Dev Network Context

[networkContext.datilDev.json](https://github.com/user-attachments/files/15908466/networkContext.datilDev.json)

### CODE REVIEW

**Comment by hwrdtm** [comment]
For now, I'm going to delegate this task to someone else and they can pick up on the PR where I've left off.

File: local-tests/setup/tinny-environment.ts [hunk 1]
```diff
@@ -317,10 +320,10 @@ export class TinnyEnvironment {
...
     }
     if (this.network === LIT_TESTNET.LOCALCHAIN) {
-      this._testnet = await this._shivaClient.startTestnetManager();
-      // wait for the testnet to be active before we start the tests.
-      await this._testnet.pollTestnetForActive();
-      await this._testnet.getTestnetConfig();
+      // this._testnet = await this._shivaClient.startTestnetManager();
+      // // wait for the testnet to be active before we start the tests.
+      // await this._testnet.pollTestnetForActive();
+      // await this._testnet.getTestnetConfig();
```
Comment (joshLong145) [inline local-tests/setup/tinny-environment.ts:326]: Have a solution for this. could you revert this change?

File: local-tests/setup/tinny-environment.ts [hunk 2]
```diff
@@ -380,7 +383,7 @@ export class TinnyEnvironment {
      * ====================================
      */
     if (this.network === LIT_TESTNET.LOCALCHAIN) {
-      const networkContext = this._testnet.ContractContext;
+      // const networkContext = this._testnet.ContractContext;
```
Comment (joshLong145) [inline local-tests/setup/tinny-environment.ts:386]: Same as above

File: local-tests/setup/tinny-person.ts [hunk 3]
```diff
@@ -101,7 +103,7 @@ export class TinnyPerson {
     //  * ====================================
     //  */
     if (this.envConfig.network === LIT_TESTNET.LOCALCHAIN) {
-      const networkContext = this.envConfig.contractContext;
+      // const networkContext = this.envConfig.contractContext;
```
Comment (joshLong145) [inline local-tests/setup/tinny-person.ts:106]: Same as above

File: local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsClaimKeys.ts [hunk 4]
```diff
@@ -15,9 +15,10 @@ import { TinnyEnvironment } from 'local-tests/setup/tinny-environment';
  * - Then: The claim operation should successfully return signatures, derived key IDs, and validate the existence and structure of claimed results.
  * *
  * Test Commands:
- * ❌ Not supported in Cayenne
```
Comment (joshLong145) [inline local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsClaimKeys.ts:18]: Why did we remove this comment?

File: local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToEncryptDecryptZip.ts [hunk 5]
```diff
@@ -9,9 +9,10 @@ import { log } from '@lit-protocol/misc';
 
 /**
  * Test Commands:
- * ❌ Not supported in Cayenne
```
Comment (joshLong145) [inline local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToEncryptDecryptZip.ts:12]: Why did we remove this comment?

File: local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsClaimMultipleKeys.ts [hunk 6]
```diff
@@ -13,9 +13,10 @@ import { TinnyEnvironment } from 'local-tests/setup/tinny-environment';
  * - Then: The claim operation should successfully return signatures, derived key IDs, and validate the existence and structure of claimed results.
  * *
  * Test Commands:
- * ❌ Not supported in Cayenne
```
Comment (joshLong145) [inline local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsClaimMultipleKeys.ts:16]: Same as above

File: local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsConsoleLog.ts [hunk 7]
```diff
@@ -7,9 +7,10 @@ import { TinnyEnvironment } from 'local-tests/setup/tinny-environment';
 
 /**
  * Test Commands:
- * ❌ Not supported on cayenne
```
Comment (joshLong145) [inline local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsConsoleLog.ts:10]: Same as above

File: local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsJsonResponse.ts [hunk 8]
```diff
@@ -5,9 +5,10 @@ import { TinnyEnvironment } from 'local-tests/setup/tinny-environment';
 
 /**
  * Test Commands:
- * ❌ Not supported on cayenne
```
Comment (joshLong145) [inline local-tests/tests/testUseValidLitActionCodeGeneratedSessionSigsToExecuteJsJsonResponse.ts:8]: Same as above

**Review by joshLong145 — COMMENTED** [review]
Looks good outside of the `local-test` changes
